### PR TITLE
Makes result classes rely on auto value.

### LINKED ISSUE #69: Consider using AutoValue for all result classes
I would like to be able to log the result objects. Implementing `toString` for each object may become cumbersome to maintain in the future, so AutoValue seams like a good fit here.
For `FingerprintDecyptionResult` [auto-value-redacted](https://github.com/square/auto-value-redacted) can be used to avoid leacage of sensetive data in logs.

### PR DESCRIPTION
**Pull request by serj-lotutovici**
Had to create a new subtype `FingerprintExpressionResult` (not convinced by the name), because auto-value prevents from creating subclasses from other subclasses.

P.S. Sorry for some re-formats, some files were using tabs instead of spaces.

Closes #69

### CODE REVIEW

File: rxfingerprint/src/main/java/com/mtramin/rxfingerprint/data/FingerprintEncryptionResult.java [hunk 1]
```diff
@@ -17,33 +17,40 @@
...
      * @param encrypted encrypted data
      */
-    public FingerprintEncryptionResult(FingerprintResult result, String message, String encrypted) {
-        super(result, message);
-        this.encrypted = encrypted;
+    public static FingerprintEncryptionResult create(FingerprintResult result, String message, String encrypted) {
+        return new AutoValue_FingerprintEncryptionResult(message, result, encrypted);
     }
 
+    @Nullable
```
Comment (Mauin) [inline rxfingerprint/src/main/java/com/mtramin/rxfingerprint/data/FingerprintEncryptionResult.java]: Should we also `@Redact` this?

**Comment by serj-lotutovici** [comment]
20% method count increase is a lot and adding auto-value later in the life of the lib is also an option, if that ever becomes necessary. On the other hand, it's hard to imagine any project that doesn't use progurad these days. 
In any case, I'm ok with both. Just tell me what you think is best at the moment.

**Comment by Mauin** [comment]
I'd go with implementing the `toString` methods manually for now. Should RxFingerprint expose more and more data classes at some point I would consider adding AutoValue to make that easier, but for now it seems to be much leaner to simply implement the method ourselves.

**Comment by Mauin** [comment]
Closing this in favor of #73

**Pull request closed without merging**